x-pack/plugins/apm/public/components/app/service_logs/index.tsx

```diff
@@ -86,20 +87,27 @@ export function ServiceLogs() {
       height={'60vh'}
       startTimestamp={moment(start).valueOf()}
       endTimestamp={moment(end).valueOf()}
-      query={getInfrastructureKQLFilter(data)}
+      query={getInfrastructureKQLFilter(data, serviceName)}
     />
   );
 }
 
 export const getInfrastructureKQLFilter = (
-  data?: APIReturnType<'GET /internal/apm/services/{serviceName}/infrastructure'>
+  data:
+    | APIReturnType<'GET /internal/apm/services/{serviceName}/infrastructure'>
+    | undefined,
```

Comment:
Any reason why I change? Seems to be the same thing

Reply:
Before the argument was optional. Now it's a required argument that can have an `undefined`  value. The distinction seems pedantic but since Typescript doesn't allow required arguments (`serviceName`) to follow optional arguments (`data`) I had to make the change (or flip the arguments around)